packages/graphql-validate-fixtures/src/index.ts

```diff
@@ -1,68 +1,168 @@
+import {getGraphQLConfig, GraphQLProjectConfig} from 'graphql-config';
+import {AST, compile, Operation} from 'graphql-tool-utilities/ast';
 import {
-  GraphQLSchema,
-  buildClientSchema,
-  Source,
-  parse,
-  concatAST,
-} from 'graphql';
-import {compile} from 'graphql-tool-utilities/ast';
+  getGraphQLFilePath,
+  getGraphQLProjects,
+} from 'graphql-tool-utilities/config';
 
 import {
+  getOperation,
   validateFixtureAgainstAST,
-  validateFixtureAgainstSchema,
   Validation,
   Fixture,
+  MissingOperationError,
 } from './validate';
 
-export interface Paths {
-  fixturePaths: string[];
-  operationPaths?: string[];
-  schemaPath: string;
-}
-
 export interface Options {
-  schemaOnly?: boolean;
+  cwd: string;
 }
 
 export interface Evaluation extends Validation {
-  fixturePath: string;
   scriptError?: Error;
 }
 
 export async function evaluateFixtures(
-  {fixturePaths, operationPaths = [], schemaPath}: Paths,
-  {schemaOnly = false}: Options = {},
+  fixturePaths: string[],
+  {cwd}: Options,
 ): Promise<Evaluation[]> {
-  let schema: GraphQLSchema;
+  const config = getGraphQLConfig(resolve(cwd));
 
-  try {
-    const schemaJSON = await readJSON(schemaPath, {encoding: 'utf8'});
-    schema = buildClientSchema(schemaJSON.data);
-  } catch (error) {
-    throw new Error(
-      `Error parsing '${schemaPath}':\n\n${error.message.replace(
-        /Syntax Error GraphQL \(.*?\) /,
-        '',
-      )}`,
-    );
+  const projectOperations = await Promise.all(
+    getGraphQLProjects(config).map(getOperationsForProject),
+  );
+
+  return runForEachFixture(fixturePaths, (fixture) =>
+    evaluateFixture(fixture, projectOperations),
+  );
+}
+
+interface FixtureOperation {
+  ast: AST;
+  error?: any;
+  operation: Operation;
+  operationName: string;
+}
+
+interface FixtureOperationError {
+  ast: AST;
+  error: any;
+  operation?: Operation;
+  operationName?: string;
+}
+
+function evaluateFixture(
+  fixture: Fixture,
+  projectOperations: GraphQLProjectOperations[],
+): Evaluation {
+  const fixtureOperations = projectOperations.map<
+    FixtureOperation | FixtureOperationError
+  >(({ast}) => {
+    try {
+      return {
+        ast,
+        ...getOperation(fixture, ast),
+      };
+    } catch (error) {
+      return {
+        ast,
+        error,
+        operation: undefined,
+        operationName: undefined,
+      };
+    }
+  });
+
+  const validFixtureOperations = fixtureOperations.filter(
+    ({operation, error}) => {
+      return operation != null && error == null;
+    },
+  ) as FixtureOperation[];
+
+  if (validFixtureOperations.length > 1) {
+    const [{operationName}] = validFixtureOperations;
+    const projectNames = projectOperations
+      .map(({config}) => config.resolveProjectName())
+      .join(', ');
+
+    return {
+      fixturePath: fixture.path,
+      validationErrors: [
+        new Error(
+          `Ambiguous operation name '${operationName}' in '${
```

Comment:
We could run through all schemas and check for dupes, but it's probably not worthwhile since we wouldn't really save much time and this should be a super rare occurrence. Right now there isn't much you can do if this happens other than change the name. Chris and I discussed the option of using a `@schema` pragma similar to the `@operation` pragma to override ambiguities, this would solve the problem pretty easily if it ever did happen. In reality, this should never happen because myself and the guidance I will provide for any other teams looking to put their schema in web will be to prefix your operations (i.e., 
`CoreCapitalIndexQuery` and `BourgeoisCapitalIndexQuery`).

Reply:
I was more concerned about readability than speed.  If this is super rare, and can be shuffled into a helper function that nobody ever looks at, all the better.

That said, not a dealbreaker.